### ISSUE TITLE
复现问题

### ISSUE THREAD

**Issue by loveego** [HCPLab-SYSU/CD-FER-Benchmark#15]
你好，我是华南师范大学的一名学生
对于您论文的复现我想请问您几个问题：
 复现时按照您的代码和数据集处理方式，从RAF-DB 到CK+ 数据集的迁移 best Accuracy 只有 78.29，无法达到85效果
您给出的docker 的image   putao3/images:py3-pytorch1.3-agra 里面是python2的 ，而且没有装pytorch

**Comment by putao537** [comment]
> 你好，我是华南师范大学的一名学生
> 对于您论文的复现我想请问您几个问题：
> 复现时按照您的代码和数据集处理方式，从RAF-DB 到CK+ 数据集的迁移 best Accuracy 只有 78.29，无法达到85效果
> 您给出的docker 的image putao3/images:py3-pytorch1.3-agra 里面是python2的 ，而且没有装pytorch

您好，方便将您的实验参数发一下吗？
另外，我给出的 docker image 应该都是 python3 的环境，你有输入过 python3 来试试吗？

**Comment by loveego** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HCPLab-SYSU/CD-FER-Benchmark"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月23日(星期四) 下午3:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HCPLab-SYSU/CD-FER-Benchmark] 复现问题 (#15)

谢谢你的回复，附件带有实验参数
另外，使用的是您的这个images，python的版本是

**Comment by putao537** [comment]
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "HCPLab-SYSU/CD-FER-Benchmark" ***@***.***&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 下午3:37 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [HCPLab-SYSU/CD-FER-Benchmark] 复现问题 (#15) 谢谢你的回复，附件带有实验参数 另外，使用的是您的这个images，python的版本是

您好，您还是在github上进行回复吧，我并没有看见您的附件。
建议截图，这样比较好排查问题。

**Comment by loveego** [comment]
参数
**Train on source**
Log Name:ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK
Output Path: .
Backbone: ResNet50
Resume Model: ../preTrainedModel/ir50_ms1m_112_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster.pkl
CUDA_VISIBLE_DEVICES: 0
================================================
Use 112 * 112 Image
SourceDataset: RAF
TargetDataset: CK+
Train Batch Size: 64
Test Batch Size: 64
================================================
Train Epoch: 60
Learning Rate: 0.000100
Momentum: 0.900000
Weight Decay: 0.000100
================================================
Number of classes : 7
Use global feature and local feature.
Use Intra GCN.
Use Inter GCN.
Use Mean in Cluster.
================================================
Building Train and Test Dataloader...
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
The test source dataset quantity: 3067
The test source dataset distribute: 329, 73, 160, 1185, 478, 162, 680
The test target dataset quantity: 129
The test target dataset distribute: 21, 9, 15, 24, 12, 15, 33
Done!
================================================
Building Model...
Resume Model: ../preTrainedModel/ir50_ms1m_112_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster.pkl
Done!
================================================
Initialize Mean in Cluster....
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
[Source Domain] Cost time : 101.123076s
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
[Target Domain] Cost time : 7.149424s
Done!
================================================
Building Optimizer...
Done!
================================================
Run Experiment...

**Transfer to Target Doamin**
Log Name: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_transferToTargetDomain_RAFtoCK
Output Path: .
Backbone: ResNet50
Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
CUDA_VISIBLE_DEVICES: 0
================================================
Use 112 * 112 Image
SourceDataset: RAF
TargetDataset: CK+
Train Batch Size: 32
Test Batch Size: 32
================================================
Train Epoch: 60
Learning Rate: 0.000100
Momentum: 0.900000
Weight Decay: 0.000100
Use DAN Loss: CDAN-E
Learning Rate(Adversarial Network): 0.001000
================================================
Number of classes : 7
Use global feature and local feature.
Use Intra GCN.
Use Inter GCN.
Use Mean in Cluster.
================================================
Building Train and Test Dataloader...
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
The test source dataset quantity: 3067
The test source dataset distribute: 329, 73, 160, 1185, 478, 162, 680
The test target dataset quantity: 129
The test target dataset distribute: 21, 9, 15, 24, 12, 15, 33
Done!
================================================
Building Model...
Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
Done!
================================================
Building Adversarial Network...
Done!
================================================
Building Optimizer...
Done!
================================================
Initialize Mean in Cluster....
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
[Source Domain] Cost time : 101.711358s
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
[Target Domain] Cost time : 7.518230s
Done!
================================================
Run Experiment...

**Comment by loveego** [comment]
Train on source
Log Name:ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK
Output Path: .
Backbone: ResNet50
Resume Model: ../preTrainedModel/ir50_ms1m_112_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster.pkl
CUDA_VISIBLE_DEVICES: 0
================================================
Use 112 * 112 Image
SourceDataset: RAF
TargetDataset: CK+
Train Batch Size: 64
Test Batch Size: 64
================================================
Train Epoch: 60
Learning Rate: 0.000100
Momentum: 0.900000
Weight Decay: 0.000100
================================================
Number of classes : 7
Use global feature and local feature.
Use Intra GCN.
Use Inter GCN.
Use Mean in Cluster.
================================================
Building Train and Test Dataloader...
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
The test source dataset quantity: 3067
The test source dataset distribute: 329, 73, 160, 1185, 478, 162, 680
The test target dataset quantity: 129
The test target dataset distribute: 21, 9, 15, 24, 12, 15, 33
Done!
================================================
Building Model...
Resume Model: ../preTrainedModel/ir50_ms1m_112_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster.pkl
Done!
================================================
Initialize Mean in Cluster....
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
[Source Domain] Cost time : 101.123076s
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
[Target Domain] Cost time : 7.149424s
Done!
================================================
Building Optimizer...
Done!
================================================
Run Experiment...

Transfer to Target Doamin
Log Name: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_transferToTargetDomain_RAFtoCK
Output Path: .
Backbone: ResNet50
Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
CUDA_VISIBLE_DEVICES: 0
================================================
Use 112 * 112 Image
SourceDataset: RAF
TargetDataset: CK+
Train Batch Size: 32
Test Batch Size: 32
================================================
Train Epoch: 60
Learning Rate: 0.000100
Momentum: 0.900000
Weight Decay: 0.000100
Use DAN Loss: CDAN-E
Learning Rate(Adversarial Network): 0.001000
================================================
Number of classes : 7
Use global feature and local feature.
Use Intra GCN.
Use Inter GCN.
Use Mean in Cluster.
================================================
Building Train and Test Dataloader...
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
The test source dataset quantity: 3067
The test source dataset distribute: 329, 73, 160, 1185, 478, 162, 680
The test target dataset quantity: 129
The test target dataset distribute: 21, 9, 15, 24, 12, 15, 33
Done!
================================================
Building Model...
Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
Done!
================================================
Building Adversarial Network...
Done!
================================================
Building Optimizer...
Done!
================================================
Initialize Mean in Cluster....
The train source dataset quantity: 12260
The train source dataset distribute: 1289, 279, 717, 4770, 1980, 704, 2521
[Source Domain] Cost time : 101.711358s
The train target dataset quantity: 1125
The train target dataset distribute: 228, 66, 162, 183, 72, 120, 294
[Target Domain] Cost time : 7.518230s
Done!
================================================
Run Experiment...

**Comment by loveego** [comment]
[实验参数.docx](https://github.com/HCPLab-SYSU/CD-FER-Benchmark/files/7216617/default.docx)

**Comment by putao537** [comment]
> [实验参数.docx](https://github.com/HCPLab-SYSU/CD-FER-Benchmark/files/7216617/default.docx)

您好，您是不是没有在 Source Domain上训练（也就是运行 TrainOnSourceDomain.py），就直接运行 TransferToTargetDomain.py了？

**Comment by loveego** [comment]
有的

**Comment by putao537** [comment]
> 有的

可是您的实验参数显示您加载的模型仍然是预训练模型（即 ResNet-50 pretrained on MS1M），而不是在 Source Domain 上训练好的 Checkpoint。

**Comment by loveego** [comment]
在tansfer to target domain.py 加载的模型是
Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
是train on source 上训练出来的

**Comment by putao537** [comment]
> 在tansfer to target domain.py 加载的模型是
> Resume Model: ResNet50_CropNet_GCNwithIntraMatrixAndInterMatrix_useCluster_withoutAFN_trainOnSourceDomain_RAFtoCK.pkl
> 是train on source 上训练出来的

抱歉，刚刚没有看见后续的东西。您在源域上的训练结果跑到多少呢？

**Comment by loveego** [comment]
在源域上最好的结果：train / test 1.0000/0.8849

**Comment by putao537** [comment]
> 在源域上最好的结果：train / test 1.0000/0.8849

抱歉，我看了参数并没找到有问题的地方（除了两次训练的 Batch Size 大小不一致，但是应该不会导致这么大的性能差异）。
您能否提供这两次训练的详细训练日志呢？

**Comment by loveego** [comment]
好的，我也尝试过修改超参数 ，batchsize为32，lr_ad 0.01 目前跑到效果最好为78.29
[复现.docx](https://github.com/HCPLab-SYSU/CD-FER-Benchmark/files/7216768/default.docx)

**Comment by putao537** [comment]
抱歉，我暂时无法判断哪里出现了错误，也暂时没有精力与时间去处理。
或许您能与已复现的其他工作的作者进行讨论，比如： "JDMAN: Joint Discriminative and Mutual Adaptation Networks for Cross-Domain Facial Expression Recognition", ACM MM 2021。

**Comment by putao537** [comment]
如果您仍然无法解决该问题，欢迎重新开启该问题。
